### DISCUSSION TITLE
Exception when caching routes if `Livewire::setUpdateRoute` have been used

### DISCUSSION THREAD

**Discussion by AndreiBiziuk** [livewire/livewire#9116] [7. Bugs (archived)]
### Livewire version

v3.5.18

### Laravel version

v11.36.1

### Which PHP version are you using?

PHP 8.3

### What is the problem?

If I change my `livewire.update` route in `web.php` and then run `php artisan route:cache', I get exception:

```bash
   LogicException 

  Unable to prepare route [cabinet/livewire/update] for serialization. Another route has already been assigned name [livewire.update].

  at vendor/laravel/framework/src/Illuminate/Routing/AbstractRouteCollection.php:248
    244▕             $route->name($this->generateRouteName());
    245▕ 
    246▕             $this->add($route);
    247▕         } elseif (! is_null($symfonyRoutes->get($name))) {
  ➜ 248▕             throw new LogicException("Unable to prepare route [{$route->uri}] for serialization. Another route has already been assigned name [{$name}].");
    249▕         }
    250▕ 
    251▕         $symfonyRoutes->add($route->getName(), $route->toSymfonyRoute());
    252▕ 

      +19 vendor frames 

  20  artisan:13
      Illuminate\Foundation\Application::handleCommand()
```

### Code snippets

Include this in `routes/web.php`
```php

Livewire::setScriptRoute(function ($handle) {
    return Route::get('/cabinet/livewire/livewire.js', $handle);
});

Livewire::setUpdateRoute(function ($handle) {
    return Route::post('/cabinet/livewire/update', $handle);
});

```

run `php artisan route:cache`

### Wirebox URL

https://wirebox.app/b/40dwz

### How do you expect it to work?

Caching of routes should work

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@AndreiBiziuk yeah you can't have two routes with the same name.

Try this

```php
Livewire::setScriptRoute(function ($handle) {
    return Route::get('/cabinet/livewire/livewire.js', $handle)->name('custom-script');
});

Livewire::setUpdateRoute(function ($handle) {
    return Route::post('/cabinet/livewire/update', $handle)->name('custom-update');
});
```

**Reply by AndreiBiziuk** [reply]
Shouldn't `setUpdateRoute` replace the existing route instead of creating a new one with the same name?
I'm trying to run Laravel with Livewire using the `'/cabinet'` path prefix. When I use both `setScriptRoute` and `setUpdateRoute` I get the desired result, but the problem occurs when I try to enable route caching. The exception is thrown only when using `setUpdateRoute`, but when using only `setScriptRoute` there is no caching issue.

**Reply by AndreiBiziuk** [reply]
If I rename routes, as you suggested, caching completes, but prefix disappears.

Before caching:

![изображение](https://github.com/user-attachments/assets/71be3a4f-48a8-4046-a020-97736416089b)
![изображение](https://github.com/user-attachments/assets/466db06c-3cc1-4047-89b7-9a705e1cb439)


After caching:

![изображение](https://github.com/user-attachments/assets/2d2d4f0d-7dc2-4b23-9e0b-a82b8f00ea5d)
![изображение](https://github.com/user-attachments/assets/63d3ab8d-70e8-48a3-8ea3-26de9f9f8ab4)

**Reply by AndreiBiziuk** [reply]
Looks like route caching ignores `setUpdateRoute` and `setScriptRoute`.
